flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -89,6 +89,17 @@
                                     + TASK_MANAGER_SERVICE_ACCOUNT.key()
                                     + "' for jobmanager and taskmanager respectively.");
 
+    public static final ConfigOption<List<Map<String, String>>> JOB_MANAGER_OWNER_REFERENCE =
+            key("kubernetes.jobmanager.owner.reference")
+                    .mapType()
+                    .asList()
+                    .noDefaultValue()
+                    .withDescription(
+                            "The user-specified owner reference to be set to the JobManager Deployment. "
+                                    + "The owner reference is used to delete deployment (and the actual cluster) when controlling resource is deleted. "
+                                    + "The value should be in the form of key:value,where keys define owner reference parameters: apiVersion, "
```

Comment:
nit:
```suggestion
                                    + "The value should be in the form of key:value, where keys define owner reference parameters: apiVersion, "
```

Reply:
I think the following two things are not very clear to me after reading the description.
- Why is the owner reference only set for JobManager. Are other resources guaranteed to be cleaned up? We might want to make it more explicit that "delete the actual cluster" means all resources are cleaned up.
- How exactly to use the configuration option. I would suggest to add an example (with multiple owner references).

Reply:
1. When deleted, job manager will clean up all all of the resources associated with the cluster, as a result, we only need to delete a job manager to delete a cluster.
2. As for multiple owner references, see here https://medium.com/@bharatnc/kubernetes-garbage-collection-781223f03c17. If an object contains more than one owner reference, it will only be deleted when both of them are deleted. Although it is possible, its rarely used. An example configuration is in KubernetesJobManagerFactoryTest.java

Reply:
Hi @blublinsky,
You don't need to explain to me, because I'm quite familiar with how these things work for Flink and Kubernetes.
What I meant in the previous comment is that, these things should also be explained in the description of the config option, so that users who may not be as familiar with Flink and Kubernetes as us won't get confused.